### PR TITLE
[22.01] Add mzqc datatype

### PR DESCRIPTION
**Pull request by bernt-matthias**
needed here https://github.com/galaxyproteomics/tools-galaxyp/pull/653

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -287,6 +287,7 @@
     <datatype extension="xquest.xml" type="galaxy.datatypes.proteomics:XquestXML" mimetype="application/xml" display_in_upload="true"/>
     <datatype extension="spec.xml" type="galaxy.datatypes.proteomics:XquestSpecXML" mimetype="application/xml" display_in_upload="true"/>
     <datatype extension="qcml" type="galaxy.datatypes.proteomics:QCML" mimetype="application/xml" display_in_upload="true"/>
+    <datatype extension="mzqc" type="galaxy.datatypes.text:Json" subclass="true" display_in_upload="true" description="Quality control data in JSON format (https://github.com/HUPO-PSI/mzQC)."/>
```
Comment (bgruening) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: MassSpec quality control data in JSON ...?

Comment (bernt-matthias) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: Sounds good. Then we should also fix this for `qcml` where the desciption has been added later (I copy pasted from this) https://github.com/galaxyproject/galaxy/blob/bbab98ab46b06b9216c07c5433c7fc1edc92f3a7/lib/galaxy/config/sample/datatypes_conf.xml.sample#L310

Will add a PR for dev.

Comment (bernt-matthias) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]:
```suggestion
    <datatype extension="mzqc" type="galaxy.datatypes.text:Json" subclass="true" display_in_upload="true" description="Mass spectrometry quality control data in JSON format (https://github.com/HUPO-PSI/mzQC)."/>
```

**Comment by mvdbeek** [comment]
22.01 has been released now, so that's the oldest release we can merge this into.

**Comment by bernt-matthias** [comment]
> 22.01 has been released now, so that's the oldest release we can merge this into.

Changed it.

**Pull request merged**